### DISCUSSION TITLE
Question about SDF lib plugin and making SDF in general

### DISCUSSION THREAD

**Discussion by benthebear93** [google-deepmind/mujoco#2182] [Asking for Help] [answered]
### Intro

Hi!

I am a graduate student at SDU, I use MuJoCo for my research on Contact manipulation.
For few days, i have been trying to get a good collision mesh for bettetr contact.

I found few ways to do it.

1. Write your own SDFs and make it as a plugin
2. Use [obj2mjcf](https://github.com/kevinzakka/obj2mjcf) (or [CoACD](https://github.com/SarahWeiii/CoACD), same thing) to decompose Non convex obj files.
3. Use "mujoco.sdf.sdflib" plugin and set geom type="sdf"

First one may be the best option to have good performance but required too many work.
Second one works pretty fine but if i want to have finner collision meshes, i needed to decompose more which makes too many obj files for one object.

Third one works good without having bunch of objs but **i couldn't really find the documents about SDF lib plugin.**

### My setup

MuJoCo 3.2.0
Python
Linux 22.04

### My question

Is third method just same plugin as first one but default? 

What are the differences between the methods mentioned above?
Thank you. 

### Confirmations

- [X] I searched the [latest documentation](https://mujoco.readthedocs.io/en/latest/overview.html) thoroughly before posting.
- [X] I searched previous [Issues](https://github.com/google-deepmind/mujoco/issues) and [Discussions](https://github.com/google-deepmind/mujoco/discussions), I am certain this has not been raised before.

**Comment by quagla** [comment, marked as answer]
Hi, the collision algorithm for methods 1 and 3 will be the same. The difference is that 1 requires writing a new plugin, while 3 means using the already existing sdflib plugin (like in this example https://github.com/google-deepmind/mujoco/blob/main/model/plugin/sdf/cow.xml) - the underlying representation of the SDF is a voxel grid computed from the mesh using this library https://github.com/UPC-ViRVIG/SdfLib

Using 3 is likely not as fast as using 1 but it's obviously more general because it's really hard to come up with an analytic SDF for complicated objects, so computing it automatically using a voxel grid is a big advantage.

**Reply by yuvaltassa** [reply]
That said, @quagla we should probably document the sdflib business somewhere?

**Reply by benthebear93** [reply]
Thanks for the information :)
